### PR TITLE
Add - tree description svg component

### LINKED ISSUE #776: Redraw this diagram with SVG
Redraw this diagram with SVG
Author: kirjs
Slide: [Local](http://localhost:4200/component-tree/description),[Public](https://angular-presentation.firebaseapp.com/component-tree/description)

### PR DESCRIPTION
**Pull request by AleksandrBu**
Added a tree description svg component

Fixes #776

### CODE REVIEW

**Comment by kirjs** [comment]
Add a screenshot?

File: apps/angular-presentation/src/app/codelabs/typescript/typescript/tree-description-svg/tree-description-svg.component.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'tree-description-svg',
+  templateUrl: './tree-description-svg.component.html',
```
Comment (kirjs) [inline apps/angular-presentation/src/app/codelabs/typescript/typescript/tree-description-svg/tree-description-svg.component.ts]: OnPush?

File: apps/angular-presentation/src/app/codelabs/typescript/typescript/tree-description-svg/index.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+export * from './tree-description-svg.component';
```
Comment (kirjs) [inline apps/angular-presentation/src/app/codelabs/typescript/typescript/tree-description-svg/index.ts]: Maybe rename it to something like `component-hierarchy-svg.component`?

File: apps/angular-presentation/src/app/codelabs/typescript/typescript/tree-description-svg/tree-description-svg.component.html [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+      style="fill:rgb(13.299561%,44.699097%,100%);stroke-width:0.72;stroke-linecap:butt;stroke-linejoin:round;stroke:rgb(0%,9.799194%,27.799988%);stroke-opacity:1;stroke-miterlimit:10;"
+      d="M 297.351562 271.21875 C 288.328125 262.800781 273.730469 262.800781 264.71875 271.21875 C 297.351562 301.679688 297.351562 301.679688 297.351562 301.679688 C 306.359375 293.261719 306.359375 279.628906 297.351562 271.21875 Z M 297.351562 271.21875 "
+      transform="matrix(1,0,0,-1,0,540)"/>
+    <path
+      style="fill:rgb(13.299561%,44.699097%,100%);stroke-width:0.72;stroke-linecap:butt;stroke-linejoin:round;stroke:rgb(0%,9.799194%,27.799988%);stroke-opacity:1;stroke-miterlimit:10;"
+      d="M 257.878906 285.058594 C 257.878906 297.878906 268.328125 308.28125 281.21875 308.28125 C 294.109375 308.28125 304.558594 297.878906 304.558594 285.058594 C 304.558594 272.238281 294.109375 261.839844 281.21875 261.839844 C 268.328125 261.839844 257.878906 272.238281 257.878906 285.058594 Z M 257.878906 285.058594 "
+      transform="matrix(1,0,0,-1,0,540)"/>
+  </g>
+
+  <text x="268" y="186" class="child-first"> Child</text>
```
Comment (kirjs) [inline apps/angular-presentation/src/app/codelabs/typescript/typescript/tree-description-svg/tree-description-svg.component.html]: I18n all the things

File: apps/angular-presentation/src/app/codelabs/typescript/typescript/typescript.module.ts [hunk 4]
```diff
@@ -42,7 +43,11 @@ const routes = RouterModule.forChild(
     SimpleEditorModule,
     PresentationModule,
   ],
-  declarations: [TypescriptComponent, TypescriptSvgComponent],
+  declarations: [
+    TypescriptComponent,
```
Comment (kirjs) [inline apps/angular-presentation/src/app/codelabs/typescript/typescript/typescript.module.ts]: this should go in `component-tree.module.ts`, not here

**Comment by AleksandrBu** [comment]
tree-description-component

**Comment by AleksandrBu** [comment]
> Add a screenshot?

Added

**Comment by NothingEverHappens** [comment]
👍 
Can you add also a screenshot with it on the actual page?

**Comment by AleksandrBu** [comment]
> 👍
> Can you add also a screenshot with it on the actual page?

Exactly there is no actual page for this svg picture) I just prepared the code, but didn't replace previous picture, which is here: https://codelab.fun/component-tree/description

**Pull request closed without merging**